Dockerfile

```diff
@@ -1,9 +1,37 @@
-FROM ubuntu:22.04
-RUN apt-get update && apt-get upgrade --yes && apt-get install git python3.9 python3-pip --yes
-ADD . /code
-WORKDIR /code
-
-RUN pip install .
-RUN ape plugins install alchemy ens etherscan foundry hardhat infura ledger solidity template tokens trezor vyper
-RUN pip install eth-rlp==0.3.0
+#---------------------------------------------------------------------------------------------
+# See LICENSE in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+ARG PYTHON_VERSION="3.9"
+ARG PLUGINS_FILE="./recommended-plugins.txt"
+
+FROM python:${PYTHON_VERSION}
+
+RUN apt-get update && apt-get upgrade --yes && apt-get install git
+
+# See http://label-schema.org for metadata schema
+# TODO: Add `build-date` and `version`
+LABEL maintainer="ApeWorX" \
+      org.label-schema.schema-version="2.0" \
+      org.label-schema.name="ape" \
+      org.label-schema.description="Ape Ethereum Framework." \
+      org.label-schema.url="https://docs.apeworx.io/ape/stable/" \
+      org.label-schema.usage="https://docs.apeworx.io/ape/stable/userguides/quickstart.html#via-docker" \
+      org.label-schema.vcs-url="https://github.com/ApeWorX/ape" \
+      org.label-schema.docker.cmd="docker run --volume $HOME/.ape:/root/.ape --volume $HOME/.vvm:/root/.vvm --volume $HOME/.solcx:/root/.solcx --volume $PWD:/root/project --workdir /root/project apeworx/ape compile"
+
+RUN useradd --create-home --shell /bin/bash harambe
+WORKDIR /home/harambe
+COPY . .
+
+RUN pip install --upgrade pip \
+    && pip install --no-cache-dir . \
+    && pip install -r recommended-plugins.txt \
```

Comment:
The recommended plugins get installed using the `pip` approach now, so now change to `ape plugins install` is needed, as @fubuloubu pointed out that was not a good idea.

I am thinking this may not be the final implementation of this, but it is closer to what we want and achieves the goals.